### ISSUE TITLE
Build a model using siamese network

### ISSUE THREAD

**Issue by gshanbhag525** [CropAi/Data-Modeling#20]
Try to implement a model using siamese network which will give better accuracy at lesser epoch

**Comment by GowthamChowta** [comment]
I would like to work on this issue

**Comment by gshanbhag525** [comment]
@gauti091 Have u worked with keras n siamese network before?

**Comment by GowthamChowta** [comment]
I have worked with keras and have never worked with siamese network before. But I have gone through this link explaining the model architecture. 
https://towardsdatascience.com/one-shot-learning-with-siamese-networks-using-keras-17f34e75bb3d

If you can provide any reference paper or blog I will go through it.

Thanks.

**Comment by gshanbhag525** [comment]
Thats good that you are interested and want to learn. Heres paper https://www.researchgate.net/publication/339620844_Using_a_Convolutional_Siamese_Network_for_Image-Based_Plant_Species_Identification_with_Small_Datasets

**Comment by abhishekgupta368** [comment]
I can work with it easily. can you please assign me

**Comment by GowthamChowta** [comment]
Thank you for sharing the reference paper.

I have gone through the paper and made the following items we need to do for our project especially for modelling.

1. Building tuples from the image data for training.
2. Defining the model architecture with a custom loss function.
3. Inference, how to use similarity score learned from the training to predict on test image.

Please correct me If I am wrong, In the following issue,
 I will only be working on defining model architecture right?

If yes, all the other items will be new issues?

**Comment by GowthamChowta** [comment]
Will the model architecture defined in this picture is fine to implement.
![1_v40QXakPBOmiq4lCKbPu8w](https://user-images.githubusercontent.com/58596028/76216453-9ffaef00-6236-11ea-8a4b-9339a728220f.png)

**Comment by GowthamChowta** [comment]
I have submitted a pull request for this issue #31

**Comment by abhishekgupta368** [comment]
sorry for the delay, I will do it now and push it, I have exams. hope you understand.

**Comment by GowthamChowta** [comment]
HI Abhishek, before starting anything can you see the file #31. If anything is needed I can help you.

**Comment by abhishekgupta368** [comment]
ok

**Comment by abhishekgupta368** [comment]
your model is incomplete

**Comment by abhishekgupta368** [comment]
i have completed but little preprocessing left, I will complete it by night of by morning

**Comment by abhishekgupta368** [comment]
can you please put a HARD tag, it requires too much preprocessing and file writing.
thank you for giving me time.

**Comment by GowthamChowta** [comment]
Abhishek,

Can you please let me know If I can help you in any way.

**Comment by abhishekgupta368** [comment]
No, I have completed, Just check my code

**Comment by abhishekgupta368** [comment]
can you please verify my code

**Comment by abhishekgupta368** [comment]
Hello, Can you please verify it?
